benches/curve_factory_benchmarks.rs

```diff
@@ -0,0 +1,244 @@
+
+    // Build stream with only Curve protocol
+    let mut stream = ProtocolStreamBuilder::new(&tycho_url, Chain::Ethereum)
+        .exchange::<EVMPoolState<PreCachedDB>>(
+            "vm:curve",
+            tvl_filter.clone(),
+            Some(curve_pool_filter),
+        )
+        .auth_key(Some(tycho_api_key))
+        .set_tokens(all_tokens)
+        .await
+        .skip_state_decode_failures(true)
+        .build()
+        .await
+        .map_err(|e| {
+            warn!("Failed to build Curve protocol stream: {:?}", e);
+            e
+        })
+        .expect("Failed to build Curve protocol stream")
+        .boxed();
+
+    // Load initial data and organize by factory
+    let mut factory_data: CurveDataByFactory = HashMap::new();
+    
+    match stream.next().await {
+        Some(Ok(message)) => {
+            info!(
+                "Successfully loaded {} Curve pairs and {} states",
+                message.new_pairs.len(),
+                message.states.len()
+            );
+
+            // Organize pools and components by factory
+            for (id, component) in message.new_pairs.iter() {
+                // Extract factory from static_attributes
+                let factory_name = get_factory_name(component);
+                
+                let factory_data = factory_data
+                    .entry(factory_name)
+                    .or_insert_with(|| CurveFactoryData {
+                        pools: HashMap::new(),
+                        components: HashMap::new(),
+                        states: HashMap::new(),
+                    });
+                    
+                factory_data.pools.insert(id.clone(), component.tokens.clone());
+                factory_data.components.insert(id.clone(), component.clone());
+            }
+
+            // Organize states by factory using the component information
+            for (id, state) in message.states.into_iter() {
+                // Find which factory this state belongs to by looking up the component
+                for (_factory_name, data) in factory_data.iter_mut() {
+                    if data.components.contains_key(&id) {
+                        data.states.insert(id.clone(), state);
+                        break;
+                    }
+                }
+            }
+        }
+        Some(Err(e)) => {
+            warn!("Error loading Curve protocol data: {:?}", e);
+        }
+        None => {
+            warn!("No data received from Curve protocol stream");
+        }
+    }
+
+    factory_data
+}
+
+fn get_factory_name(component: &ProtocolComponent) -> String {
+    // Try to get factory from static_attributes
+    if let Some(factory_bytes) = component.static_attributes.get("factory") {
+        // Convert Bytes to string, taking first 8 chars for readability
+        let factory_hex = String::from_utf8(factory_bytes.to_vec()).expect("failed parsing factory address");
```

Comment:
Where are you seeing that the factory address is utf8 encoded?
From [here](https://github.com/propeller-heads/tycho-protocol-sdk/blob/main/substreams/ethereum-curve/src/pool_factories.rs) it seems to be encoded like this

```
/// Converts address bytes into a Vec<u8> containing a leading `0x`.
fn address_to_bytes_with_0x(address: &[u8; 20]) -> Vec<u8> {
    address_to_string_with_0x(address).into_bytes()
}

/// Converts address bytes into a string containing a leading `0x`.
fn address_to_string_with_0x(address: &[u8]) -> String {
    format!("0x{}", hex::encode(address))
}

...

Attribute {
    name: "factory".into(),
    value: address_to_bytes_with_0x(&CRYPTO_POOL_FACTORY),
    change: ChangeType::Creation.into(),
},
```